### PR TITLE
added the chatbot

### PR DESCRIPTION
**Pull request by Sarthak5598**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -27,6 +30,17 @@
...
 client.chat_postMessage(channel=DEPLOYS_CHANNEL_NAME, text="bot started v1.9 240611-1 top")
 
+template = """
+    You're a Software Engineer (Mentor) at OWASP,
+    Your job is to provide help to contributors with a short message.
+    Contributor' Question :{Doubt}
+"""
+
+
+OpenAI_client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))
```
Comment (arkid15r) [inline app.py:40]: Could you follow [PEP8 naming conventions](https://peps.python.org/pep-0008/#naming-conventions)?

**Review by arkid15r — COMMENTED** [review]
Just a general question before reviewing it -- as far as I remember we agreed this bot should have greeting functionality only. Has it changed recently?

File: app.py (hunk 1 above)
Comment (Sarthak5598) [reply app.py:40]: Sure !

**Comment by Sarthak5598** [comment]
I think with the greeting we also want the AI bot in this @DonnieBLT right?

**Comment by Sarthak5598** [comment]
Okay, I will add logs to this , and when the logs are enough we will use them to train the AI

File: app.py [hunk 2]
```diff
@@ -140,13 +153,61 @@ def handle_message(payload):
...
-        user = message["user"]  # The user ID of the person who sent the message
-        text = message.get("text", "")  # The text of the message
+        doubt = message.get("text", "")
+        prompt = template.format(doubt=doubt)
+
+        today = datetime.now(timezone.utc).date()
+        rate_limit_key = f"global_daily_request_{today}"
+        total_token_used = cache.get(rate_limit_key, 0)
+
+        if len(doubt) > 50:
```
Comment (arkid15r) [inline app.py:183]: To be consistent w/ the error message:
```suggestion
        if len(doubt) >= 50:
```

Also consider not using a magic number -- create a descriptively named variable for that.

Comment (Sarthak5598) [reply app.py:183]: Sure!! thanks for the suggestion

**Comment by Sarthak5598** [comment]
@arkid15r Please take a final look ! , thanks !

**Comment by arkid15r** [comment]
> @arkid15r Please take a final look ! , thanks !

👍

**Review by arkid15r — COMMENTED** [review]
👍

**Pull request merged**